platform/admin/admin-configuration-configupdater/src/main/java/org/codice/ddf/admin/configuration/ConfigurationUpdater.java

```diff
@@ -0,0 +1,322 @@
+ * properly updates the etc directory.
+ *
+ * <p>This configuration installer handles <b>writeback</b> like the Felix one did and does a few
+ * other things:
+ *
+ * <ul>
+ *   <li>Removes files from etc when they are deleted from {@link
+ *       org.osgi.service.cm.ConfigurationAdmin}
+ *   <li>Protects against tampering with the {@code felix.fileinstall.filename} by third parties,
+ *       which could disrupt the dropping of config files
+ *   <li>Ensures password fields in config files are encrypted
+ * </ul>
+ *
+ * Handling <b>writeback</b> means running this class handles the updating of dropped-in config
+ * files in the etc directory. To keep Felix itself from disrupting this process, {@code
+ * enableConfigSave} in {@code custom.properties} should be set to {@code false}.
+ */
+public class ConfigurationUpdater implements ConfigurationPersistencePlugin {
+  private static final Logger LOGGER = LoggerFactory.getLogger(ConfigurationUpdater.class);
+
+  private static final Pattern ENC_PATTERN = Pattern.compile("^ENC\\((.*)\\)$");
+
+  static final String FELIX_FILENAME = "felix.fileinstall.filename";
+
+  private final ConfigurationAdmin ddfConfigAdmin;
+
+  private final List<PersistenceStrategy> strategies;
+
+  private final EncryptionService encryptionService;
+
+  private final Map<String, CachedConfigData> pidDataMap;
+
+  /*
+   * SortedServiceList implements java.util.List using generic parameter <T> and not a concrete type,
+   * so the ctor cannot be matched by the blueprint container if we did List<PersistenceStrategy>
+   *
+   * See https://issues.apache.org/jira/browse/ARIES-960
+   */
+  public ConfigurationUpdater(
+      ConfigurationAdmin ddfConfigAdmin, List strategies, EncryptionService encryptionService) {
+    this.ddfConfigAdmin = ddfConfigAdmin;
+    this.strategies = strategies;
+    this.encryptionService = encryptionService;
+    this.pidDataMap = new ConcurrentHashMap<>();
+  }
+
+  /** @return a read-only map of configuration pids to felix file install properties. */
+  @VisibleForTesting
+  Map<String, CachedConfigData> getPidDataMap() {
+    return Collections.unmodifiableMap(pidDataMap);
+  }
+
+  /**
+   * @inheritDoc
+   * @throws IOException if an error occurs reading configuration data
+   * @throws InvalidSyntaxException technically impossible since {@code null} is being passed as the
+   *     configuration filter. See {@link ConfigurationAdmin#listConfigurations(String)}.
+   * @see ConfigurationPersistencePlugin
+   */
+  @Override
+  public void initialize(ConfigurationContextFactory factory) {
+    Configuration[] configs;
+    try {
+      configs = ddfConfigAdmin.listConfigurations(null);
+    } catch (IOException | InvalidSyntaxException e) {
+      LOGGER.error("Problem fetching configurations. ", e);
+      return;
+    }
+    if (configs == null) {
+      return;
+    }
+    pidDataMap.putAll(
```

Comment:
❗️ We need to process each config objects we find in cfg admin as if it was an update so we can update etc right away. This will account for a race condition where either changes to config objects occurred before your bundle got initialized or again if files with un-encrypted passwords were dropped in etc before we start, the config installer would have processed them and updated cfg admin with them before you are initialize. These should be saved back to etc with encrypted passwords at initialization otherwise they will never be encrypted until the next update happens on these objects and that might never happen.